app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java

```diff
@@ -914,7 +914,11 @@ public static boolean isDataStale() {
     }
 
     public static BgReading readingNearTimeStamp(double startTime) {
-        final double margin = (4 * 60*1000);
+        double margin = (4 * 60*1000);
+        if(DexCollectionType.hasLibre()) {
```

Comment:
`hasLibre()` is too broad a condition at this point. There are many places in the code base where features are expecting the minimum period between readings to be 300 seconds. For example a limitter device might emit the same reading twice within a 5 minute window due to a re-transmission.  Like the rest of these changes, I think this should be tied to: `external_blukon_algorithm` preference item.  Possibly even then this could introduce unexpected results, for example with low prediction etc.